WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.m

```diff
@@ -616,17 +616,19 @@ - (void)tableViewDidChangeContent:(UITableView *)tableView
 
 - (void)tableView:(UITableView *)tableView willDisplayCell:(UITableViewCell *)cell forRowAtIndexPath:(NSIndexPath *)indexPath
 {
-    // If searching, don't try and sync additional posts.
-    if (self.isSearching) {
-        return;
-    }
-    // Are we approaching the end of the table?
-    if ((indexPath.section + 1 == self.tableView.numberOfSections) &&
-        (indexPath.row + PostsLoadMoreThreshold >= [self.tableView numberOfRowsInSection:indexPath.section])) {
-
-        // Only 3 rows till the end of table
-        if ([self currentPostListFilter].hasMore) {
-            [self.syncHelper syncMoreContent];
+    if ([self isViewOnScreen]) {
```

Comment:
Nitpicky, but rather than nesting all the things, maybe just check for the a false condition and return?